### PR TITLE
Uvcdat boxfill custom clean up

### PR DESCRIPTION
**Pull request by doutriaux1**
Fixes issues with range generation, there is still a weird issue about a gm that becomes vistrails.core sort of instance but I will let @remram44 and al take a look at this in a separate issue. I suspect this is due to the extensions having changed from "string" default to "boolean" (hence we get back "False" rather than a real Python False) but why did the gm disappeared is the real issue left. Anyway it doesn't prevent this to work, just something we should eventually look into.

### CODE REVIEW

**Comment by doutriaux1** [comment]
this will need https://github.com/UV-CDAT/uvcdat/pull/1221

**Comment by aashish24** [comment]
@doutriaux1 who should review this branch? Also, how do we verify if things are working correctly?

**Comment by doutriaux1** [comment]
@aashish24 @remram44 yes please review, as the PR says it fix the way ranges are generated. Click Edit. Create a boxfill, select "Custom" for the type. And set min/max/number of level, click apply. Play around by turning off ext1/ext2. The idea is that ext1/ext2 should be turned off before generating ranges (no way to know if levels/colors were auto-generated or user input).

**Comment by doutriaux1** [comment]
did anybody take a look at this today?

File: vistrails/gui/uvcdat/graphicsMethodsWidgets.py [hunk 1]
```diff
@@ -504,13 +497,20 @@ def generateRanges(self):
 
             # Gen colors (Log)
             numIntervals = len(values) - 1
-            d = int(222 / (numIntervals - 1))
-            for a in range(numIntervals):
-                colors.append(16 + a * d)
             
         if self.includeZeroButtonGroup.isChecked('On'):
             values.insert(0, 0.0)
+        if str(self.ext1ButtonGroup.buttonGroup.button(self.ext1ButtonGroup.buttonGroup.checkedId()).text()).lower()[0]=="y":
```
Comment (aashish24) [inline vistrails/gui/uvcdat/graphicsMethodsWidgets.py:503]: @doutriaux1 please wrap long lines. Long lines are very hard to read even on my high res display and could lead to bugs easily.

File: vistrails/packages/uvcdat_cdms/init.py [hunk 2]
```diff
@@ -1406,12 +1405,12 @@ def updateContents(self, inputPorts, fromToolBar=False):
             cgm = self.get_graphics_method(plot.plot_type, plot.graphics_method_name)
 #            cgm.setProvenanceHandler( plot.processParameterUpdate )
             if plot.graphics_method_name != 'default':
+                #cgm.list()
```
Comment (aashish24) [inline vistrails/packages/uvcdat_cdms/init.py:1408]: nice.. good to remove dead code! :clap:

File: vistrails/gui/uvcdat/graphicsMethodsWidgets.py [hunk 3]
```diff
@@ -2058,11 +2059,18 @@ def initValues(self, gm=None):
...
+        ## So we need to test against this!
+        ## But we should really figure out why the type is changed to this vistrails thing
+        ## someone with a better understanding a VisTrails inners should try
+        ## to figure this out (@remram44 @vvpalav or @hgohil2805 ???
+        if gm.ext_1 in [False,0,"n", "False"]:
             self.ext1ButtonGroup.setChecked('No')
         else:
             self.ext1ButtonGroup.setChecked('Yes')
-        if gm.ext_2 == "n":
+        if gm.ext_2 in [False,0,"n","False"]:
```
Comment (aashish24) [inline vistrails/gui/uvcdat/graphicsMethodsWidgets.py:2073]: so gm.ext_2 can have three states?

**Comment by aashish24** [comment]
:-1: Sorry, I was still able to make it go blank after I merged this change. I followed your instructions before and after the merge.

File: vistrails/gui/uvcdat/graphicsMethodsWidgets.py (hunk 3 above)
Comment (doutriaux1) [inline vistrails/gui/uvcdat/graphicsMethodsWidgets.py:2073]: shoul just be False or True but to be safe I kept all valid "user input" in. the "False" is due to Vistrails it is not even a valid value but somehow when coming from Vistrails it has this value...

**Comment by doutriaux1** [comment]
@aashish24 yes there is still case where it goes blank, nothing we can do about it. That's where I was hoping that https://github.com/UV-CDAT/VisTrails/pull/42 would be in because you would know why. If extensions are off and then you autogenerate levels and then you turn on some extensions it will fails and go blank... I can't think of a way to prevent this. Maybe turn off ability to change extensions once levels are triggered, but then how do you go back to enabling them?

**Comment by aashish24** [comment]
Okay, so do you still want to merge this one in? I am not sure entirely that this made it any better because I ended up with blank before and after but may be it was little hard before than after. @doutriaux1 let me know.

**Comment by aashish24** [comment]
after talking to @doutriaux1 merging it :+1:

**Comment by aashish24** [comment]
Just to add more info:

The branch made it harder to come to a bad state. It didn't fix the issue completely but it got better and that's why I have merged it.

**Pull request merged**